domains/page.go

```diff
@@ -0,0 +1,23 @@
+package domains
+
+// ExclusiveStartKeyPagedResponse is object to use when returning database records in paged format using Exclusive Start Key paging
+type ExclusiveStartKeyPagedResponse[Content any, LastEvaluatedKeyType any] struct {
```

Comment:
maybe this should be:
`ExclusiveStartKeyPage` or `PageWithExclusiveStartKey`

and the struct `ExclusiveStartKeyPage` should be named as `ExclusiveStartKeyPageInfo`?